test/gurka/test_build_admin.cc

```diff
@@ -0,0 +1,286 @@
+#include <gtest/gtest.h>
+
+#include "baldr/admin.h"
+#include "filesystem.h"
+#include "gurka.h"
+#include "mjolnir/admin.h"
+#include "mjolnir/adminbuilder.h"
+#include "mjolnir/pbfadminparser.h"
+#include "mjolnir/pbfgraphparser.h"
+#include "test/test.h"
+
+using namespace valhalla;
+using namespace valhalla::baldr;
+using namespace valhalla::gurka;
+using namespace valhalla::mjolnir;
+
+namespace {
+
+// GetAdminInfo() requires a sqlite db handle and tiles. This creates a mock
+// graph with two states part of the same country - with a highway between them.
+valhalla::gurka::map BuildPBF(const std::string& workdir) {
+  const std::string ascii_map = R"(
+        A-------B-------C-------I-------J
+        |       |       |   W   |       |
+        |       |       |   |   |       |
+        |   G---|---H---|---X---|---Y   |
+        |       |       |   |   |       |
+        |       |       |   Z   |       |
+        F-------E-------D-------L-------K
+  )";
+
+  // To define an administrative boundary, the nodes must form a closed polygon.
+  const gurka::ways ways = {{"ABCDEFA", {}},
+                            {"ABEFA", {}},
+                            {"BCDEB", {}},
+                            {"CIJKLDC", {}},
+                            {"CILDC", {}},
+                            {"IJKLI", {}},
+                            {"GH",
+                             {
+                                 {"highway", "primary"},
+                             }},
+                            {"HX",
+                             {
+                                 {"highway", "primary"},
+                             }},
+                            {"XY",
+                             {
+                                 {"highway", "primary"},
+                             }},
+                            {"WX",
+                             {
+                                 {"highway", "primary"},
+                             }},
+                            {"XZ",
+                             {
+                                 {"highway", "primary"},
+                             }}};
+
+  // X lives Japan which allows named intersections - and is named.
+  // gurka automatically names the nodes by their name in the ascii map
+  // if you want to make sure there is no name you need to send empty string
+  const gurka::nodes nodes = {
+      {"X", {{"junction", "named"}, {"name", "namae wa desu ka"}}},
+      {"Y", {{"junction", "yes"}, {"name", ""}}},
+  };
```

Comment:
@ktatterso i made this explicit. gurka does stuff for us automatically but that ends up masking the behavior we wanted to test a little bit. i found setting it directly here is easier to reason about (beacuse you dont have to know that the gurka method is actually always adding a name to every node)